demo/site/src/pages/api/preview.ts

```diff
@@ -0,0 +1,31 @@
+import createGraphQLClient, { GraphQLClientOptions } from "@src/util/createGraphQLClient";
+import { gql } from "graphql-request";
+
+import { GQLValidateSitePreviewHashQuery, GQLValidateSitePreviewHashQueryVariables } from "./preview.generated";
+
+export default async function handler(req, res) {
+    const data = await createGraphQLClient().request<GQLValidateSitePreviewHashQuery, GQLValidateSitePreviewHashQueryVariables>(
+        gql`
+            query ValidateSitePreviewHash($timestamp: Float!, $hash: String!) {
+                validateSitePreviewHash(timestamp: $timestamp, hash: $hash)
+            }
+        `,
+        {
+            timestamp: parseInt(req.query.timestamp),
+            hash: req.query.hash,
+        },
+    );
+    if (!data.validateSitePreviewHash) {
+        return res.status(401).json({ message: "Validation failed" });
+    }
+
+    const previewData: PreviewData = {
+        includeInvisiblePages: true,
+        includeInvisibleBlocks: req.query.includeInvisibleBlocks === "true",
+        previewDamUrls: true,
+    };
+    res.setPreviewData(previewData);
```

Comment:
I'm not sure if this is an intended use case for preview data. Also, doesn't this mean I need to get a new hash every time the user toggles the "Include invisible content" button in the admin?

Reply:
That's how it works currently. See my other answer in https://github.com/vivid-planet/comet/pull/1237#issuecomment-1753528554 - I tried other solutions but none was as simple as storing this data in setPreviewData.